### ISSUE TITLE
Make all modules replaceable

### ISSUE THREAD

**Issue by 34j** [voicepaw/so-vits-svc-fork#533]
**Is your feature request related to a problem? Please describe.**
Renovate the config file and make all modules replacable or optional

- Support TTS
- SSL
  - HuBERT
  - HuBERT Chinese
  - ContentVec
  - [DistilHuBERT](https://arxiv.org/abs/2110.01900)
  - Whisper
- Vocoder
	- NSF-HifiGan
	- BIGVGan
	- PC-DDSP
- One shot (so-vits-svc-5.0)
- NaturalSpeech (so-vits-svc-5.0)
- F0 Decoder
- Flow
	- ResidualFlow
	- NanoFlow
	- Flow++?
- F0 mean filtering
- top1 search (rvc)
- (Add HifiSinger, DiffSinger etc.)

**Describe the solution you'd like**
Someone else will oddly accomplish all of this. Or another repository will accomplish all of this and make this repository unnecessary.

**Additional context**
I will never do this.

**Comment by Kickbub** [comment]
In that case will this repo be renamed?

**Comment by 34j** [comment]
Yes, I quite like fish-diffusion in VC related repos, but I could not touch their code due to some somewhat concerning parts.

**Comment by avachon100510** [comment]
I added another module to replace: FreeSimpleGUI would replace PySimpleGUI.

**Comment by avachon100510** [comment]
Consider checking it out: https://github.com/voicepaw/so-vits-svc-fork/issues/1243
